taxi/views.py

```diff
@@ -26,94 +26,154 @@ def index(request):
         "num_manufacturers": num_manufacturers,
         "num_visits": num_visits + 1,
     }
+    return render(request, "taxi/index.html",
+                  context=context)
+
+
+class BaseListView(LoginRequiredMixin,
+                   generic.ListView):
+    paginate_by = 5
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        query = self.request.GET.get("q")
+        if query:
+            queryset = queryset.filter(self.get_search_filter(query))
+        print(queryset)
+        return queryset
+
+    def get_search_filter(self, query):
+        raise NotImplementedError("Override this method in child classes")
+
+
+class DriverListView(BaseListView):
+    model = Driver
+    context_object_name = "driver_list"
+    template_name = "taxi/driver_list.html"
+    paginate_by = 10
+    ordering = ["username"]
 
-    return render(request, "taxi/index.html", context=context)
+    def get_search_filter(self, query):
+        return Q(username__icontains=query)
 
 
-class ManufacturerListView(LoginRequiredMixin, generic.ListView):
+class CarListView(BaseListView):
+    model = Car
+    queryset = Car.objects.select_related("manufacturer")
+    template_name = "taxi/car_list.html"
+
+    def get_search_filter(self, query):
+        return Q(model__icontains=query)
+
+
+def get_search_filter(query):
+    return Q(name__icontains=query)
+
+
+class ManufacturerListView(ListView):
     model = Manufacturer
     context_object_name = "manufacturer_list"
     template_name = "taxi/manufacturer_list.html"
-    paginate_by = 5
+    paginate_by = 10
+    ordering = ["name"]
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        query = self.request.GET.get("q")
+        if query:
+            queryset = queryset.filter(get_search_filter(query))
```

Comment:
The function 'get_search_filter' used here is defined outside of the class, which is inconsistent with the other ListView classes. Consider defining 'get_search_filter' within the 'ManufacturerListView' class for consistency and to avoid potential issues with function scope.